src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadata/CompilationUtility.cs

```diff
@@ -54,6 +56,35 @@ public static Compilation CreateCompilationFromVBCode(string code)
             }
         }
 
+        //TODO: Only process CSharp assembly currently
+        public static Compilation CreateCompilationFromAssembly(IEnumerable<string> assemblyPaths)
+        {
+            try
+            {
+                var paths = assemblyPaths.ToList();
+                //TODO: "mscorlib" should be ignored while extracting metadata from .NET Core/.NET Framework
+                paths.Add(typeof(object).Assembly.Location);
+                var assemblies = (from path in paths
+                                  select MetadataReference.CreateFromFile(path)).ToList();
+                var complilation = CS.CSharpCompilation.Create("EmptyProjectWithAssembly", new SyntaxTree[] { }, assemblies);
+                return complilation;
+            }
+            catch (Exception e)
+            {
+                Logger.Log(LogLevel.Warning, $"Error generating compilation from assemblies {string.Join(Environment.NewLine, assemblyPaths)}: {e.Message}. Ignored.");
```

Comment:
yes, you're right, I should return null here.

---
In reply to: [95291502](https://github.com/dotnet/docfx/pull/1131#discussion_r95291502) [](ancestors = 95291502,95290045,95289758)